compiler/rustc_lint/src/context.rs

```diff
@@ -983,7 +983,7 @@ impl<'tcx> LateContext<'tcx> {
                     return Ok(path);
                 }
 
-                path.push(Symbol::intern(&disambiguated_data.data.to_string()));
+                path.push(Symbol::intern(&disambiguated_data.to_string()));
```

Comment:
Is this change needed for this PR? If so we must be careful not to break things because the format of these names might be something end users rely on.

Reply:
I don't believe that it is strictly neccesary - I was trying to standardize the way that types were emitted, but I'll revert these changes to minimize the diff.